enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java

```diff
@@ -66,4 +77,75 @@ public static EndpointCluster processEndpoints(org.wso2.choreo.connect.discovery
+        }
+
+        if (mockedApiConfig.getIn().equalsIgnoreCase(APIConstants.MockApiConstants.HEADER)) {
+            setMockApiResponse(responseObject, headersMap, mockedApiConfig, acceptType, false);
+        } else if (mockedApiConfig.getIn().equalsIgnoreCase(APIConstants.MockApiConstants.QUERY)) {
+            Map<String, String> queryParamMap = requestContext.getQueryParameters();
+            setMockApiResponse(responseObject, queryParamMap, mockedApiConfig, acceptType, false);
+        } else {
+            setMockApiResponse(responseObject, null, mockedApiConfig, acceptType, true);
+        }
+    }
+
+    /**
+     * Prepares mock API response considering the properties map and mockedApiConfig.
+     *
+     * @param responseObject  Response object (represents response for the API call)
+     * @param propertiesMap   A map which includes values specified in headers or query parameters
+     * @param mockedApiConfig Holds the JSON values specified in the mock API implementation
+     * @param acceptType      Denotes accepted contend type as the response
+     */
+    private static void setMockApiResponse(ResponseObject responseObject, Map<String, String> propertiesMap,
+                                           MockedApiConfig mockedApiConfig, String acceptType, boolean isDefault) {
+        String requestValuePosition = mockedApiConfig.getIn();
+        String nameFieldInJson;
+
+        // condition handles case-sensitiveness of query parameters.
+        if (requestValuePosition.equalsIgnoreCase(APIConstants.MockApiConstants.HEADER)) {
+            nameFieldInJson = mockedApiConfig.getName().toLowerCase();
+        } else {
+            nameFieldInJson = mockedApiConfig.getName();
+        }
+
+        if (!isDefault && (propertiesMap == null || !propertiesMap.containsKey(nameFieldInJson))) {
+            log.error("Response determining value not available in the mock API request.");
+            return;
+        }
+        List<MockedResponseConfig> responseConfigList = mockedApiConfig.getResponses();
+        // iterates over the mock API responses.
+        for (MockedResponseConfig responseConfig : responseConfigList) {
+            responseObject.setStatusCode(responseConfig.getCode());
+            Map<String, String> headerMap = new HashMap<>();
+            // iterates over the headers list in the mock API JSON.
+            for (MockedHeaderConfig header : responseConfig.getHeaders()) {
+                headerMap.put(header.getName(), header.getValue());
+            }
+            // checks and assigns the accepted content type
+            MockedContentConfig content = responseConfig.getContent();
+            if (content.getContentMap().containsKey(acceptType)) {
+                headerMap.put(APIConstants.CONTENT_TYPE_HEADER, acceptType);
+                responseObject.setMockApiResponseContent(content.getContentMap().get(acceptType));
+            } else {
```

Comment:
what will be the response to client if acceptType is not in the responses list and there is not application/json response defined also?

Reply:
In the docs I am expecting to say we need to define at least a default JSON message to work with mock API implementations. Otherwise we cannot provide a response to the users. Code is written to provide JSON value as the default response message.

Reply:
+1 for add to docs, BTW I think we should support the error scenario @praminda mentioned above
for eg: HTTP 501 or HTTP 503 and an error message